todo_app/app.py

```diff
@@ -30,60 +29,156 @@ def __init__(self, id):
 def create_app():
     app = Flask(__name__)
     app.config.from_object('todo_app.flask_config.Config')
+    app.logger.setLevel(app.config['LOG_LEVEL'])
     app.wsgi_app = ReverseProxied(app.wsgi_app)
 
-    #os.environ['OAUTHLIB_INSECURE_TRANSPORT']
-    gh_blueprint = make_github_blueprint(client_id=os.environ['GITHUB_CLIENT_ID'], client_secret=os.environ['GITHUB_CLIENT_SECRET'])
-    app.register_blueprint(gh_blueprint, url_prefix='/github_login')
+    if app.config['LOGGLY_TOKEN'] is not None:
+        handler = HTTPSHandler(f'https://logs-01.loggly.com/inputs/{app.config["LOGGLY_TOKEN"]}/tag/todo-app')
+        formatter = jsonlogger.JsonFormatter("[%(asctime)s] %(levelname)s in %(module)s: %(message)s %(requesterIpAddr)s")
+        handler.setFormatter(formatter)
+        app.logger.addHandler(handler)
 
+    client_id = os.environ['GITHUB_CLIENT_ID']
+    client_secret = os.environ['GITHUB_CLIENT_SECRET']
+    base_url="https://api.github.com"
+    authorization_url="https://github.com/login/oauth/authorize"
+    token_endpoint = "https://github.com/login/oauth/access_token"
+
+    client = WebApplicationClient(client_id)
     todo = mongodb_todo()
-    usermanager = mongodb_user()
+    usermanager = MongoDbUserService()
 
     login_manager = LoginManager()
     login_manager.init_app(app)
 
+
+    @app.errorhandler(Exception)
+    def handle_error(e):
+        app.logger.error('exception', extra={
+            "method": "{}".format(request.method),
+            "requesterIpAddr": "{}".format(request.remote_addr),
+            "url": "{}".format(request.url)
+        }, exc_info=True)
+
+        return render_template("error.html", error=str(e))
+    
+    @app.after_request
+    def after_request(response):
+        app.logger.info("after_request", extra={
+            "method": "{}".format(request.method),
+            "requesterIpAddr": "{}".format(request.remote_addr),
+            "url": "{}".format(request.url),
+            "status_code": "{}".format(response.status)
+        })
+        return response
+
     @login_manager.user_loader
     def load_user(user_id):        
         return UserToLogin(user_id)
     
     @login_manager.unauthorized_handler
     def unauthenticated():
-        return redirect(url_for('github_login'))
+        app.logger.info("Unauthorized attemp made.", extra={
+            "method": "{}".format(request.method),
+            "requesterIpAddr": "{}".format(request.remote_addr),
+            "url": "{}".format(request.url)
+        })
+        return redirect(url_for('login'))
     
     @app.route('/logout')
     @login_required
     def logout():
+        app.logger.info("User {} logged out of the system.".format(current_user.id), extra={
+            "method": "{}".format(request.method),
+            "requesterIpAddr": "{}".format(request.remote_addr),
+            "url": "{}".format(request.url)
+        })
```

Comment:
Minor, but this "extra" section is quite similar between lots of your logging - a good opportunity to extract that out into a function!